SonarQube.Client.Tests/Models/ServerSentEvents/SqServerSentEventParserTests.cs

```diff
@@ -0,0 +1,191 @@
+        public void Parse_InvalidEventData_MissingEventDataField_Null()
+        {
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(new[]{ "event: some type" });
+
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Parse_InvalidEventData_EventDataIsEmpty_Null()
+        {
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(new[] { "event: some type", "data: " });
+
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Parse_InvalidEventData_EventDataIsNotInCorrectFormat_Null()
+        {
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(new[] { "event: some type", "data : extra space" });
+
+            result.Should().BeNull();
+        }
+
+
+        [TestMethod]
+        public void Parse_CorrectEventString_ParsedEvent()
+        {
+            var eventLines = new[]
+            {
+                "event: some event type",
+                "data: some event data"
+            };
+
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(eventLines);
+
+            result.Should().NotBeNull();
+            result.Type.Should().Be("some event type");
+            result.Data.Should().Be("some event data");
+        }
+
+        [TestMethod]
+        public void Parse_CorrectEventString_MultilineData_ParsedEvent()
+        {
+            var eventLines = new[]
+            {
+                "event: some event type",
+                "data: some event data1",
+                "data: ",
+                "data: some event data2",
+            };
+
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(eventLines);
+
+            result.Should().NotBeNull();
+            result.Type.Should().Be("some event type");
+            result.Data.Should().Be("some event data1some event data2");
+        }
+
+        [TestMethod]
+        public void Parse_HasJunkFields_JunkFieldsIgnored()
+        {
+            var eventLines = new[]
+            {
+                "junk1: junk field1",
+                "EVENT: junk event type",
+                "event:",
+                "event: some event type",
+                "data: ",
+                "data: some event data1",
+                "junk2: junk field2",
+                "DATA: junk data2",
+                "data: some event data2",
+                "junk3: junk field3"
+            };
+
+            var testSubject = CreateTestSubject();
+
+            var result = testSubject.Parse(eventLines);
+
+            result.Should().NotBeNull();
+            result.Type.Should().Be("some event type");
+            result.Data.Should().Be("some event data1some event data2");
+        }
+
+        [TestMethod]
+        public void Parse_EventTypeIsNotTheFirstField_EventTypeIsStillParsedCorrectly()
```

Comment:
I was thinking about enforcing that the first item is always the type, but if there are junk fields, we would need to clean the data first and ensure that it's the first _valid_ field. I think that's why they decided not to do it on the java side, just to reduce complexity.

Reply:
oh ok, then keep it like this